src/openforms/js/components/admin/form_design/RegistrationFields.stories.js

```diff
@@ -0,0 +1,389 @@
+    ],
+  },
+};
+
+export const ObjectsAPI = {
+  args: {
+    configuredBackends: [
+      {
+        key: 'backend4',
+        name: 'Objects API',
+        backend: 'objects_api',
+        options: {
+          version: 2,
+          objectsApiGroup: 1,
+          objecttype: 'd89f3a0e-096b-45ea-afe1-ce211d63d1f2',
+          objecttypeVersion: 1,
+          updateExistingObject: false,
+          informatieobjecttypeSubmissionReport: '',
+          uploadSubmissionCsv: false,
+          informatieobjecttypeSubmissionCsv: '',
+          informatieobjecttypeAttachment: '',
+          organisatieRsin: '',
+          variablesMapping: [],
+          geometryVariableKey: '',
+        },
+      },
+    ],
+  },
+
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+
+    await userEvent.click(canvas.getByRole('button', {name: 'Opties instellen'}));
+
+    // check that the sole api group is automatically selected
+    const apiGroupSelect = await screen.findByLabelText('API-groep');
+    expect(apiGroupSelect).toHaveValue('1');
+
+    await waitFor(() => {
+      expect(screen.getByLabelText('Objecttype')).toHaveValue(
+        'd89f3a0e-096b-45ea-afe1-ce211d63d1f2'
+      );
+    });
+
+    await waitFor(() => {
+      expect(screen.getByLabelText('Versie')).toHaveValue('1');
+    });
+  },
+};
+
+export const ObjectsAPILegacy = {
+  name: 'Objects API (legacy)',
+  args: {
+    configuredBackends: [
+      {
+        key: 'backend4',
+        name: 'Objects API',
+        backend: 'objects_api',
+        options: {
+          version: 1,
+          objectsApiGroup: 1,
+          objecttype: 'd89f3a0e-096b-45ea-afe1-ce211d63d1f2',
+          objecttypeVersion: 1,
+          updateExistingObject: false,
+          productaanvraagType: 'productAanvraag',
+          informatieobjecttypeSubmissionReport: 'https://example.com/api/v1/iot/123',
+          uploadSubmissionCsv: true,
+          informatieobjecttypeSubmissionCsv: 'https://example.com/api/v1/iot/123',
+          informatieobjecttypeAttachment: 'https://example.com/api/v1/iot/123',
+          organisatieRsin: '123456782',
+          contentJson: '{% json_summary %}',
+          paymentStatusUpdateJson: '{% json_summary %}',
+        },
+      },
+    ],
+  },
+
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+
+    await userEvent.click(canvas.getByRole('button', {name: 'Opties instellen'}));
+
+    // check that the sole api group is automatically selected
+    const apiGroupSelect = await screen.findByLabelText('API-groep');
+    expect(apiGroupSelect).toHaveValue('1');
+
+    await waitFor(() => {
+      expect(screen.getByLabelText('Objecttype')).toHaveValue(
+        'd89f3a0e-096b-45ea-afe1-ce211d63d1f2'
+      );
+    });
+
+    await waitFor(() => {
+      expect(screen.getByLabelText('Versie')).toHaveValue('1');
+    });
+  },
+};
+
+export const ZGW = {
```

Comment:
scope and time - this PR is already big enough as it is. ZGW will follow in due time, but I really need PRs to be manageable so that branches are short-lived.